src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketPostClient.java

```diff
@@ -0,0 +1,21 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import okhttp3.Response;
+
+/**
+ * Basic Bitbucket post client. A client is normally threadsafe and can be used multiple times.
+ *
+ * @param <T> The model used in the post parameters
+ */
+public interface BitbucketPostClient<T> {
+
+    /**
+     * Make the call out to Bitbucket and read the response.
+     *
+     * TODO: Fill out params and describe the output and error possibilities, waiting on makePost implementation
+     *
+     * @param data
+     * @return
+     */
+    Response post(T data);
```

Comment:
Need to clarify with @goravjoshi if this is the correct return type

Reply:
Have a look at [this](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/blob/gjoshi/fetch-webhooks-details/src/main/java/com/atlassian/bitbucket/jenkins/internal/client/HttpRequestExecutor.java) modified HttpRequestExecutor. 
Get and Post should not be Bitbucket specific, they are just HTTP Get and Post. The client factory then needs to use this to make a call to Bitbucket server.